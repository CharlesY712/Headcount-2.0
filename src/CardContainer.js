import React from 'react';
import Card from './Card';
import './CardContainer.css';

const CardContainer = ({stats, displayComparedCards}) => {
  const districtCards = stats.map((district, index) => {
    return <Card 
      district={district} 
      key={index}
    />;
  });
  
  return (
    <div 
      onClick={displayComparedCards} className='card-container'>
      {districtCards}
    </div>
  );
};

export default CardContainer;