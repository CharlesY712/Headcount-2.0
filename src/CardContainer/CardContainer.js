import React from 'react';
import Card from '../Card/Card';
import './CardContainer.css';
import PropTypes from 'prop-types';

const CardContainer = ({stats, displayComparedCards, compareArray}) => {
  const districtCards = stats.map((district, index) => {
    return <Card 
      district={district} 
      key={index}
      displayComparedCards={displayComparedCards}
      compareArray={compareArray}
    />;
  });
  
  return (
    <div className='card-container'>
      {districtCards}
    </div>
  );
};

CardContainer.propTypes = {
  stats: PropTypes.array,
  displayComparedCards: PropTypes.func,
  compareArray: PropTypes.array
};

export default CardContainer;