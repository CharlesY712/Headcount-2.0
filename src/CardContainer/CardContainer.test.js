import React from 'react';
import CardContainer from './CardContainer';
import { shallow, mount } from 'enzyme';

describe('CardContainer', () => {

  const mockArray = [
    {
      location: 'MEEKER RE1',
      stats:
        {
          '2004': 0,
          '2005': 0.505
        }
    }
  ];

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<CardContainer stats = {mockArray}/>);
  });

  it('should exist', () => {
    expect(wrapper).toBeDefined();
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});