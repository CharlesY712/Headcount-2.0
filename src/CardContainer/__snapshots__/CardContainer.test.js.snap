// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    stats={
        Array [
            Object {
              "location": "MEEKER RE1",
              "stats": Object {
                "2004": 0,
                "2005": 0.505,
              },
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          compareArray={undefined}
          displayComparedCards={undefined}
          district={
                    Object {
                              "location": "MEEKER RE1",
                              "stats": Object {
                                "2004": 0,
                                "2005": 0.505,
                              },
                            }
          }
/>,
      ],
      "className": "card-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "compareArray": undefined,
          "displayComparedCards": undefined,
          "district": Object {
            "location": "MEEKER RE1",
            "stats": Object {
              "2004": 0,
              "2005": 0.505,
            },
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            compareArray={undefined}
            displayComparedCards={undefined}
            district={
                        Object {
                                    "location": "MEEKER RE1",
                                    "stats": Object {
                                      "2004": 0,
                                      "2005": 0.505,
                                    },
                                  }
            }
/>,
        ],
        "className": "card-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "compareArray": undefined,
            "displayComparedCards": undefined,
            "district": Object {
              "location": "MEEKER RE1",
              "stats": Object {
                "2004": 0,
                "2005": 0.505,
              },
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
